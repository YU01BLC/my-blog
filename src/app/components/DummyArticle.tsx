import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';

export default function DummyArticle() {
  return (
    <Card>
      <CardHeader title='記事タイトル(dummy)' subheader='2023/12/1' />
      <CardContent>
        <CardMedia component='img' height='194' image='/assets/dummy-blog-image.jpg' alt='Paella dish' />
        <Typography variant='body1' color='text.secondary' sx={{ paddingTop: '1em' }}>
          ダミー記事です。
          <br />
          右上の「記事を書く」ボタンから新規記事を執筆してください。
        </Typography>
        <Typography variant='body1' color='text.secondary' sx={{ paddingTop: '1em' }}>
          ※ &thinsp; 記事投稿後ダミー記事は非表示となります。
        </Typography>
      </CardContent>
    </Card>
  );
}
