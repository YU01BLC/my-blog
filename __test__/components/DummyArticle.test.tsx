import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import DummyArticle from '../../src/app/components/DummyArticle';

describe('DummyArticleコンポーネントSnapshot', () => {
  it('snapshot', () => {
    const { asFragment } = render(<DummyArticle />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DummyArticleコンポーネント表示確認', () => {
  it('DummyArticleコンポーネントが表示されていること', () => {
    render(<DummyArticle />);
    const element = screen.getByTestId('dummy-article');
    expect(element).toBeInTheDocument();
  });
});

describe('Cardコンポーネント表示確認', () => {
  it('CardHeaderコンポーネントが表示されていること', () => {
    render(<DummyArticle />);
    const cardElement = screen.getByText('記事タイトル(dummy)');
    expect(cardElement).toBeInTheDocument();
  });
  it('CardMediaコンポーネントが表示されていること', () => {
    render(<DummyArticle />);
    const cardElement = screen.getByAltText(/Paella dish/i);
    expect(cardElement).toBeInTheDocument();
  });
  it('Typographyコンポーネントが表示されていること', () => {
    render(<DummyArticle />);
    const cardElement = screen.getByTestId('dummy-article--text');
    expect(cardElement).toBeInTheDocument();
  });
});
