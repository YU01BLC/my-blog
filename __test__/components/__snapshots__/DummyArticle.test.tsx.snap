// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DummyArticleコンポーネントSnapshot snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-testid="dummy-article"
  >
    <div
      class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
        >
          記事タイトル(dummy)
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
        >
          2023/12/1
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <img
        alt="Paella dish"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="194"
        src="/assets/dummy-blog-image.jpg"
      />
      <p
        class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
      >
        ダミー記事です。
        <br />
        右上の「記事を書く」ボタンから新規記事を執筆してください。
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
        data-testid="dummy-article--text"
      >
        ※   記事投稿後ダミー記事は非表示となります。
      </p>
    </div>
  </div>
</DocumentFragment>
`;
